### PR TITLE
:sparkles: Update service worker skipWaiting workflow

### PR DESCRIPTION
**Pull request by kefranabg**
See #134 

This SW workflow is much better to handle the "new content available" workflow

Thanks @NOtherDev for the suggestion :pray: I'd love to get your review !

### CODE REVIEW

File: src/misc/register-service-worker.js [hunk 1]
```diff
@@ -30,3 +30,12 @@ if (process.env.NODE_ENV === 'production') {
     }
   })
 }
+
+let refreshing = false
```
Comment (tbetous) [inline src/misc/register-service-worker.js]: I do not understand this part. When is `refreshing` set to true ?

**Comment by tbetous** [comment]
Good job, I think I have understood a major part of your PR. There are some part that needs explanations. Great job for the test part.

File: src/misc/register-service-worker.js (hunk 1 above)
Comment (kefranabg) [reply src/misc/register-service-worker.js]: When a new Service Worker take the control ;)

File: src/store/app/app.actions.js [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+
+  /**
+   * Trigger service worker skipWating so the new service worker can take over.
+   * This will also trigger a window refresh (see /src/misc/register-service-worker.js)
+   */
+  serviceWorkerSkipWaiting({ state, commit }) {
+    if (isNil(state.SWRegistrationForNewContent)) return
+
+    commit('setRefreshingApp', true)
+    state.SWRegistrationForNewContent.waiting.postMessage('skipWaiting')
```
Comment (tbetous) [inline src/store/app/app.actions.js]: Is this the part that send 'skipWaiting' to the service worker ? Where can I find a registration structure, I can't find it. The `waiting` part seems to get ou from space.

Comment (kefranabg) [reply src/store/app/app.actions.js]: > Is this the part that send 'skipWaiting' to the service worker ?
Yes 

> Where can I find a registration structure, I can't find it. The waiting part seems to get ou from space.
`register-service-worker.js`

**Comment by kefranabg** [comment]
As the Service Worker worflow change in this PR, the bento-starter demo ([https://bento-starter.firebaseapp.com](https://bento-starter.firebaseapp.com)) new content available workflow will break if have the previous version on your device. To fix this, juste close the tab/kill app instance and reopen bento-starter demo, so the new service worker can take over.

**Pull request merged**